pkg/controller/dmaas_backup_controller.go

```diff
@@ -14,66 +14,88 @@ limitations under the License.
 	informers "github.com/mayadata-io/dmaas-operator/pkg/generated/informers/externalversions/mayadata.io/v1alpha1"
+	"github.com/pkg/errors"
 
+	"github.com/mayadata-io/dmaas-operator/pkg/dmaasbackup"
+	dmaaslister "github.com/mayadata-io/dmaas-operator/pkg/generated/listers/mayadata.io/v1alpha1"
 	"github.com/sirupsen/logrus"
-	velero "github.com/vmware-tanzu/velero/pkg/generated/clientset/versioned"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/labels"
+	types "k8s.io/apimachinery/pkg/types"
 	apimachineryclock "k8s.io/apimachinery/pkg/util/clock"
-	"k8s.io/client-go/kubernetes"
 	"k8s.io/client-go/tools/cache"
 )
 
 var (
+	// backupReconcilePeriod defines the interval at which updated dmaasbackup will be reconciled
+	backupReconcilePeriod = 10 * time.Second
+
+	// backupSyncPeriod defines the interval at which all the dmaasbackups will be reconciled
 	backupSyncPeriod = 30 * time.Second
 )
 
 type dmaasBackupController struct {
 	*controller
-	namespace        string
-	openebsNamespace string
-	veleroNamespace  string
-	kubeClient       kubernetes.Interface
-	dmaasClient      clientset.Interface
-	veleroClient     velero.Interface
-	clock            apimachineryclock.Clock
+	namespace   string
+	dmaasClient clientset.Interface
+	lister      dmaaslister.DMaaSBackupLister
+	backupper   dmaasbackup.DMaaSBackupper
+	clock       apimachineryclock.Clock
 }
 
 // NewDMaaSBackupController returns controller for dmaasBackup resource
 func NewDMaaSBackupController(
-	namespace, openebsNamespace, veleroNamespace string,
-	kubeClient kubernetes.Interface,
+	namespace string,
 	dmaasClient clientset.Interface,
 	dmaasBackupInformer informers.DMaaSBackupInformer,
-	veleroClient velero.Interface,
+	backupper dmaasbackup.DMaaSBackupper,
 	logger logrus.FieldLogger,
 	clock apimachineryclock.Clock,
 	numWorker int,
 ) Controller {
 	c := &dmaasBackupController{
-		controller:       newController("dmaasBackup", logger, numWorker),
-		namespace:        namespace,
-		openebsNamespace: openebsNamespace,
-		veleroNamespace:  veleroNamespace,
-		kubeClient:       kubeClient,
-		dmaasClient:      dmaasClient,
-		veleroClient:     veleroClient,
-		clock:            clock,
+		controller:  newController("dmaasBackup", logger, numWorker),
+		namespace:   namespace,
+		dmaasClient: dmaasClient,
+		backupper:   backupper,
+		lister:      dmaasBackupInformer.Lister(),
+		clock:       clock,
 	}
 
 	c.reconcile = c.processBackup
 	c.syncPeriod = backupSyncPeriod
+	c.reconcilePeriod = backupReconcilePeriod
+	c.sync = c.syncAll
+
+	shouldHandleEvent := func(obj interface{}) bool {
+		dbkp, ok := obj.(v1alpha1.DMaaSBackup)
+		if !ok {
+			return false
+		}
+
+		yes, _ := shouldProcessDMaaSBackup(dbkp)
+		return yes
+	}
 
 	dmaasBackupInformer.Informer().AddEventHandler(
 		cache.ResourceEventHandlerFuncs{
 			AddFunc: func(obj interface{}) {
-				c.enqueue(obj)
+				if ok := shouldHandleEvent(obj); ok {
```

Comment:
can we avoid the check here? (or) inside the processSchedule()?

Reply:
ok. I'll skip check here.